File: native/android/app/src/main/java/com/mediconnect/nativeapp/JourneyDemoWorkspace.kt
package com.mediconnect.nativeapp

import androidx.activity.compose.BackHandler
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalFocusManager
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.*
import androidx.compose.ui.unit.dp

@Composable
fun JourneyDemoWorkspace(journey: NativeJourney, state: DemoState, change: (DemoState) -> Unit, onHome: () -> Unit) {
    val content = journey.demo
    val copy = content.copy
    val sample = content.sample
    var guide by remember { mutableStateOf<String?>(null) }
    val focus = LocalFocusManager.current
    if (guide != null) {
        JourneyLibrary(journey, EditorialKind.KNOWLEDGE, initialSlug = guide) { guide = null }
        return
    }
    BackHandler { if (state.visit) change(state.copy(visit = false)) else onHome() }
    Scaffold { inset ->
        LazyColumn(Modifier.fillMaxSize().padding(inset).testTag("demo-workspace"),
            contentPadding = PaddingValues(24.dp), verticalArrangement = Arrangement.spacedBy(20.dp)) {
            item { TextButton(onClick = onHome) { Text(journey.home) } }
            item {
                Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                    Text(copy.eyebrow)
                    DemoHeading(when (state.role) { DemoRole.PATIENT -> copy.patient; DemoRole.DOCTOR -> copy.doctor; DemoRole.STAFF -> copy.staff })
                    Button(onClick = { focus.clearFocus(); change(state.reset(content)) }) { Text(copy.reset) }
                }
            }
            item {
                Column(Modifier.semantics { contentDescription = content.workspaceLabel }) {
                    content.roles.forEach { role ->
                        FilterChip(state.role == role.id, onClick = { focus.clearFocus(); change(state.copy(role = role.id)) }, label = { Text(role.label) })
                    }
                }
            }
            item { Text(copy.notice) }
            if (state.visit) {
                item { Button(onClick = { change(state.copy(visit = false)) }) { Text(copy.close) } }
                item { JourneyPoster(journey.homePoster, journey.imageDescription) }
                item { Text(if (state.role == DemoRole.DOCTOR) sample.patient else sample.doctor); DemoHeading(copy.callTitle) }
                item {
                    Column {
                        FilterChip(state.mic, onClick = { change(state.copy(mic = !state.mic)) }, label = { Text(if (state.mic) copy.micOn else copy.micOff) })
                        FilterChip(state.camera, onClick = { change(state.copy(camera = !state.camera)) }, label = { Text(if (state.camera) copy.camOn else copy.camOff) })
                    }
                }
                item { Text(copy.callNotice) }
            } else {
                item {
                    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                        Text(if (state.role == DemoRole.DOCTOR) copy.queue else copy.appointment)
                        Text(if (state.role == DemoRole.PATIENT) sample.doctorInitials else sample.patientInitials)
                        DemoHeading(if (state.role == DemoRole.PATIENT) sample.doctor else sample.patient)
                        Text(sample.appointment + " · " + if (state.role == DemoRole.STAFF) sample.doctor else content.tagline)
                        Text(sample.date + ", " + sample.year); Text(sample.timezone)
                        Text(state.slot, Modifier.testTag("demo-selected-slot"))
                        Text(copy.choose)
                        sample.slots.forEach { slot ->
                            FilterChip(state.slot == slot, onClick = { change(state.selectSlot(slot, content)) }, label = { Text(slot) })
                        }
                        Text(if (state.ready) copy.ready else copy.pending, Modifier.semantics { liveRegion = LiveRegionMode.Polite })
                        if (state.role == DemoRole.STAFF) Button(onClick = { change(state.copy(ready = !state.ready)) }) { Text(copy.toggle) }
                        else Button(onClick = { focus.clearFocus(); change(state.copy(visit = true)) }) { Text(copy.consult) }
                    }
                }
                item {
                    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                        if (state.role == DemoRole.STAFF) {
                            val article = requireNotNull(journey.editorial.article(EditorialKind.KNOWLEDGE, copy.coordinationArticleSlug))
                            Text(journey.editorial.knowledgeLink); DemoHeading(article.title); Text(article.summary)
                        } else {
                            Text(copy.followup); DemoHeading(copy.notes)
                            if (state.role == DemoRole.DOCTOR) Text(state.note, Modifier.testTag("demo-saved-note"))
                            else {
                                OutlinedTextField(state.draft, onValueChange = { change(state.editNote(it, content)) },
                                    label = { Text(copy.notes) }, modifier = Modifier.fillMaxWidth().testTag("demo-note"), minLines = 4)
                                Text(copy.noteHint)
                                Button(onClick = { focus.clearFocus(); change(state.saveNote()) }) { Text(copy.save) }
                                if (state.saved) Text(copy.saved, Modifier.semantics { liveRegion = LiveRegionMode.Polite })
                            }
                            Text(copy.followupBody)
                        }
                        Button(onClick = { focus.clearFocus(); guide = if (state.role == DemoRole.STAFF) copy.coordinationArticleSlug else copy.preparationArticleSlug }) {
                            Text(journey.editorial.knowledge.read)
                        }
                    }
                }
            }
        }
    }
}

@Composable private fun DemoHeading(value: String) {
    Text(value, style = MaterialTheme.typography.headlineSmall, modifier = Modifier.semantics { heading() })
}

File: native/ios/Sources/JourneyDemoWorkspace.swift
import SwiftUI

struct JourneyDemoWorkspace: View {
    let journey: NativeJourney
    let content: MobileContent
    @Binding var state: DemoState
    @Environment(\.dismiss) private var dismiss
    @FocusState private var editing: Bool
    private var demo: NativeDemo { journey.demo }
    private var copy: DemoCopy { demo.copy }
    private var sample: DemoSample { demo.sample }
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                Button(journey.home) { dismiss() }
                Text(copy.eyebrow)
                heading(state.role == .staff ? copy.staff : state.role == .doctor ? copy.doctor : copy.patient)
                Button(copy.reset) { editing = false; state.reset(demo) }.buttonStyle(.bordered)
                VStack(alignment: .leading) {
                    ForEach(demo.roles) { role in
                        Button(role.label) { editing = false; state.role = role.id }
                            .buttonStyle(.bordered)
                            .accessibilityAddTraits(state.role == role.id ? .isSelected : [])
                    }
                }.accessibilityElement(children: .contain).accessibilityLabel(demo.workspaceLabel)
                Text(copy.notice).font(.footnote)
                if state.visit { consultation } else { appointment; preparation }
            }.padding(24).frame(maxWidth: .infinity, alignment: .leading)
                .fixedSize(horizontal: false, vertical: true)
        }
        .background(content.color("background")).foregroundStyle(content.color("foreground"))
        .tint(content.color("accent"))
        .navigationTitle(demo.workspaceLabel).navigationBarTitleDisplayMode(.inline)
        .accessibilityIdentifier("demo-workspace")
    }
    private var consultation: some View {
        VStack(alignment: .leading, spacing: 20) {
            Button(copy.close) { state.visit = false }.buttonStyle(.bordered)
            if let image = JourneyPoster.image(journey.homePoster) {
                Image(uiImage: image).resizable().aspectRatio(contentMode: .fit)
                    .clipShape(RoundedRectangle(cornerRadius: 24)).accessibilityLabel(journey.imageDescription)
            } else { Text(journey.imageDescription) }
            Text(state.role == .doctor ? sample.patient : sample.doctor)
            heading(copy.callTitle)
            Toggle(state.mic ? copy.micOn : copy.micOff, isOn: $state.mic)
            Toggle(state.camera ? copy.camOn : copy.camOff, isOn: $state.camera)
            Text(copy.callNotice).font(.footnote)
        }
    }
    private var appointment: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(state.role == .doctor ? copy.queue : copy.appointment)
            Text(state.role == .patient ? sample.doctorInitials : sample.patientInitials)
            heading(state.role == .patient ? sample.doctor : sample.patient)
            Text(sample.appointment + " · " + (state.role == .staff ? sample.doctor : demo.tagline))
            Text(sample.date + ", " + sample.year); Text(sample.timezone)
            Text(state.slot).accessibilityIdentifier("demo-selected-slot")
            Text(copy.choose)
            ForEach(sample.slots, id: \.self) { slot in
                Button(slot) { try? state.selectSlot(slot, content: demo) }.buttonStyle(.bordered)
                    .accessibilityAddTraits(state.slot == slot ? .isSelected : [])
            }
            Text(state.ready ? copy.ready : copy.pending).accessibilityIdentifier("demo-readiness")
            if state.role == .staff {
                Button(copy.toggle) { state.ready.toggle() }.buttonStyle(.bordered)
            } else {
                Button(copy.consult) { editing = false; state.visit = true }.buttonStyle(.borderedProminent)
            }
        }
    }
    private var preparation: some View {
        VStack(alignment: .leading, spacing: 12) {
            if state.role == .staff {
                if let article = journey.editorial.article(.knowledge, slug: copy.coordinationArticleSlug) {
                    Text(journey.editorial.knowledgeLink); heading(article.title); Text(article.summary)
                }
            } else {
                Text(copy.followup); heading(copy.notes)
                if state.role == .doctor {
                    Text(state.note).accessibilityIdentifier("demo-saved-note")
                } else {
                    TextField(copy.notes, text: Binding(get: { state.draft }, set: { state.editNote($0, content: demo) }), axis: .vertical)
                        .lineLimit(4...8).textFieldStyle(.roundedBorder).focused($editing)
                        .accessibilityIdentifier("demo-note")
                    Text(copy.noteHint).font(.footnote)
                    Button(copy.save) { editing = false; state.saveNote() }.buttonStyle(.bordered)
                    if state.saved { Text(copy.saved).accessibilityIdentifier("demo-note-saved") }
                }
                Text(copy.followupBody)
            }
            NavigationLink(journey.editorial.knowledge.read) {
                JourneyArticle(journey: journey, content: content, kind: .knowledge,
                               slug: state.role == .staff ? copy.coordinationArticleSlug : copy.preparationArticleSlug)
            }.simultaneousGesture(TapGesture().onEnded { editing = false })
        }
    }
    private func heading(_ value: String) -> some View {
        Text(value).font(.title2).accessibilityAddTraits(.isHeader)
    }
}
